data/lua/core.lua

```diff
@@ -232,6 +232,17 @@ if wesnoth.kernel_type() == "Game Lua Kernel" then
 	wml.variable.set = wesnoth.set_variable
 	wml.variable.get_all = wesnoth.get_all_vars
 
+	local wesnoth_get_variable = wesnoth.get_variable
+	local wesnoth_set_variable = wesnoth.set_variable
+	wesnoth.variables = setmetatable({}, {
+		__index = function(_, key)
+			return wesnoth_get_variable(key)
+		end,
+		__newindex = function(_, key, value)
+			wesnoth_set_variable(key, value)
```

Comment:
Does this handle all Lua data types: nil, boolean, number, string, function, userdata, thread and table?Including the sub-types integer, float, full- and light-userdata? And strings with embedded nulls?

If so, do the __all__ values survive the round trip intact?

If not, are those that do not filtered and does an appropriate diagnostic message appear?

I like your code, as is, and would not make changes to it. If there are deficiencies, most of them should be handled in `wesnoth_set_variable()`. The only change I would make, here, would be to special-case error handling for types/values which `wesnoth_set_variable()` properly handles but you want to specifically exclude, here.

Reply:
@GregoryLundberg hmm, well, it's a syntactical sugar around `wesnoth.set_variable`, as you correctly wrote, so it handles everything that set_variable handles. I tried the following types:

* nil. Clears the variable, as per docs
* boolean. Saves and loads correctly
* number. Saves and loads correctly
* function. wesnoth.set_variable forbids saving
* table. Saves and loads correctly if the table is a correct WML table.
* string with "nulls". Saves and loads correctly (but I haven't tested game save-loads)

There are no additional pitfalls or special cases for saving them (syntax-wise).

Overall, it seems like a sane behavior already, without the need of an additional layer. What do you think?

Reply:
I wanted to feel sure you're considered such things and you've done so. Thanks.

Reply:
Yeah, sure.